Zend/tests/bug76439.phpt

```diff
@@ -0,0 +1,68 @@
+--TEST--
+Bug #76439: Don't always strip leading whitespace from heredoc T_ENCAPSED_AND_WHITESPACE tokens
+--FILE--
+<?php
+
+$foo = 1;
+
+var_dump(<<<BAR
+ $foo-
+ BAR);
+
+var_dump(<<<BAR
+ $foo -
+ BAR);
+
+var_dump(<<<BAR
+ $foo	-
+ BAR);
+
+var_dump(<<<BAR
+ $foo-$foo
+ BAR);
+
+var_dump(<<<BAR
+ $foo-$foo-
+ BAR);
+
+var_dump(<<<BAR
+ $foo-$foo-$foo
+ BAR);
+
+var_dump(<<<BAR
+ $foo
+ -
+ BAR);
+
+var_dump(<<<BAR
+ $foo
+  -
+ BAR);
+
+var_dump(<<<BAR
+$foo
+BAR);
+
+var_dump(<<<BAR
+ -
+ BAR);
+
+var_dump(<<<BAR
+  -
+ BAR);
+
```

Comment:
It might be worthwhile to print some additional labels to make it easier to debug a failing test.

Reply:
Yeah, I was a little lazy here. Updated it now.